hypothesis-python/src/hypothesis/_strategies.py

```diff
@@ -463,7 +463,8 @@ def floats(
     Attempting to exclude an endpoint which is None will raise an error;
     use ``allow_infinity=False`` to generate finite floats.  You can however
     use e.g. ``min_value=float("-inf"), exclude_min=True`` to exclude only
-    one infinite endpoint.
+    one infinite endpoint.  Because ``-0.0 == 0.0``, excluding either will
+    also exclude the other.
```

Comment:
Hmm. That seems like the wrong choice. In every other way for the `floats` API we treat `-0.0` and `0.0` as distinct. So e.g. `floats(min_value=0.0)` will *not* include `-0.0`

Reply:
Hmm, can do - we'll just need to special-case it in `next_up`.

Reply:
Thinking about it a bit more I think we're going to get confusing behaviour either way. 😢  e.g. I can easily see someone writing `floats(max_value=0.0, exclude_max=True)` and being surprised when they get `-0.0`. I still think treating them as distinct is the right thing to do, but it'll require careful documentation.

Reply:
We did not previously treat `floats(0.0, -0.0).example()` as an error.  Yay... deprecation case incoming.

Reply:
computers are hard. 😭

Reply:
Yeah.  "Fixed" now though - the 5.0 release is going to make this so much nicer when we can rely on endpoints being exactly representable as floats.